src/lib/TelemetryClient.ts

```diff
@@ -349,7 +349,22 @@ class TelemetryClient {
      * then the specified props will be updated.  Otherwise, a new instance will returned. 
```

Comment:
Just curious as to the reason for overwriting passed-in values (as opposed to the general JavaScript convention that passed in values are not overwritten).